src/main/java/com/couchbase/lite/replicator/BulkDownloader.java

```diff
@@ -134,47 +136,45 @@ protected void executeRequest(HttpClient httpClient, HttpUriRequest request) {
-                                    try {
-                                        // decompress if contentEncoding is gzip
-                                        if (Utils.isGzip(entity)) {
-                                            gzipStream = new GZIPInputStream(inputStream);
-                                            fullBody = Manager.getObjectMapper().readValue(
-                                                    gzipStream, Object.class);
-                                        } else {
-                                            fullBody = Manager.getObjectMapper().readValue(
-                                                    inputStream, Object.class);
+
+                                    // multipart
+                                    if (contentTypeHeader.getValue().contains("multipart/")) {
+                                        Log.v(TAG, "contentTypeHeader = %s",
+                                                contentTypeHeader.getValue());
+                                        _topReader = new MultipartReader(
+                                                contentTypeHeader.getValue(), this);
+                                        byte[] buffer = new byte[BUF_LEN];
+                                        int nBytesRead = 0;
+                                        while ((nBytesRead = inputStream.read(buffer)) != -1) {
+                                            _topReader.appendData(buffer, 0, nBytesRead);
                                         }
+                                        _topReader.finished();
                                         respondWithResult(fullBody, error, response);
-                                    } finally {
-                                        try {
-                                            if (gzipStream != null) {
-                                                gzipStream.close();
-                                            }
-                                        } catch (IOException e) {
-                                        }
-                                        gzipStream = null;
                                     }
+                                    // non-multipart
+                                    else {
+                                        Log.v(TAG, "contentTypeHeader is not multipart = %s",
+                                                contentTypeHeader.getValue());
+                                        fullBody = Manager.getObjectMapper().readValue(
+                                                inputStream, Object.class);
+                                        respondWithResult(fullBody, error, response);
+                                    }
+                                }
+                            } finally {
+                                try {
+                                    if (gzipStream != null) {
+                                        gzipStream.close();
```

Comment:
In line 146, gzipStream was assigned to inputStream variable, the inputStream will be closed in the outer finally block below. So closing gzipStream here is redundant.